krun/platform.py

```diff
@@ -1057,26 +1058,26 @@ def _check_rmsr_loaded(self):
         stdout, stderr, rc = run_shell_cmd(cmd, failure_fatal=True)
         debug("rmsr loaded successfully.")
 
-    def _check_aslr_disabled(self):
-        debug("Checking ASLR is off")
-        with open(self.ASLR_FILE, "r") as fh:
-                enabled = fh.read().strip()
-        if enabled == "0":
-            return  # OK!
-        else:
-            # ASLR is off, but we can try to enable it
-            debug("Turning ASLR off")
-            cmd = "%s sh -c 'echo 0 > %s'" % \
-                (self.change_user_cmd, self.ASLR_FILE)
-            stdout, stderr, rc = run_shell_cmd(cmd, failure_fatal=False)
+    def _check_aslr_enabled(self):
+        """Check ASLR is cranked to level 2"""
 
-            if rc != 0:
-                msg = "ASLR disabled (%s, expect '0' got '%s').\n" % \
-                    (self.ASLR_FILE, enabled)
-                msg += "Krun tried to turn it off, but failed."
-                fatal(msg)
+        debug("Checking ASLR is on")
+        for rep in 1, 2:
+            with open(LinuxPlatform.ASLR_FILE, "r") as fh:
+                val = fh.read().strip()
+            if val == str(LinuxPlatform.ASLR_MODE):
+                return  # OK!
             else:
-                self._check_aslr_disabled()  # should work this time
+                if rep == 2:
```

Comment:
Not easily, I could split out the opening of the file into a new method which could be monkeypatched. Is it worth the effort?